soroban-env-host/src/host_object.rs

```diff
@@ -18,11 +20,74 @@ use super::{
     U64Object, Val, VecObject,
 };
 
+/// A ticket is a number that represents permission to read a set of host
```

Comment:
Maybe I am missing something... but because objects have to be passed in as argument in order to be visible, this seems maybe over complicated? I *think* the only thing to keep track of is how deep in the stack something was shared.

Reply:
No, it's unfortunately much more complicated than that. As a minimal example speaking to your question: if A calls B(x), then x should be readable by B, sure; but if B returns and A then calls C(y), x should _not_ be readable by C, even though C is "at the same call depth" as B was.